### PR TITLE
Handle escaped quotes correctly

### LINKED ISSUE #41: Parser fails to split lines when text contains escaped quotes at the end
Simple example
```csv
id
""""""
"test"
```

Expected result
```javascript
{ id: '""' }
{ id: 'test' }
```

Actual result
```javascript
{ id: '"""\n"test' }
```

### PR DESCRIPTION
**Pull request by Gobie**
Fixes #41

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -57,9 +58,16 @@ Parser.prototype._transform = function (data, enc, cb) {
   }
 
   for (var i = start; i < buf.length; i++) {
-    // non-escaped quote (quoting the cell)
-    if (buf[i] === this.quote && (i === start || (i > start && buf[i - 1] !== this.escape))) {
-      this._quoted = !this._quoted
+    if (!this._escaped && buf[i] === this.escape && buf[i + 1] === this.quote && i !== start) {
```
Comment (mafintosh) [inline index.js]: could you check that i + i < buf.length as well? otherwise we're gonna get a deopt from v8

**Comment by mafintosh** [comment]
LGTM except for my comment

File: index.js (hunk 1 above)
Comment (Gobie) [inline index.js]: Yeah, sure.
`buf[i + 1]` lower in this function doesn't cause the same issue?

Comment (mafintosh) [inline index.js]: oh good catch. we should guard that one as well

**Comment by Gobie** [comment]
@mafintosh Fixed deoptimization

**Comment by mafintosh** [comment]
nice solution

**Comment by calvinmetcalf** [comment]
ping on publishing this?

**Comment by mafintosh** [comment]
1.9.2

**Pull request merged**